add(pack): Azure SQL Managed Instance (#639)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/cloud-azure-database-sqlmanagedinstance.md

apps from on-premises to a fully managed and secured PaaS cloud environment 
with automatic patching and version updates, automated backups and high 
availability.

The Centreon Plugin-Pack *Azure SQL Managed Instance* can rely on Azure API or Azure CLI 
to collect the metrics related to the SQL Managed Instance service.

## Plugin Pack Assets

### Monitored Objects

* Azure *SQL Managed Instance* instances

### Discovery rules

The Centreon Plugin-Pack *Azure SQL Managed Instance* includes a Host Discovery *provider* to automatically discover the Azure instances of a given
subscription and add them to the Centreon configuration.
This provider is named **Microsoft Azure SQL Managed Instance**:

![image](../../../assets/integrations/plugin-packs/procedures/cloud-azure-database-sqlmanagedinstance-provider.png)

> This discovery feature is only compatible with the 'api' custom mode. 'azcli' is not supported yet.

More information about the Host Discovery module is available in the Centreon documentation:
[Host Discovery](../../../monitoring/discovery/hosts-discovery.html)

### Collected Metrics

<!--DOCUSAURUS_CODE_TABS-->

<!--Cpu-->

> | Metric Name                                   | Description            | Unit |
> |:----------------------------------------------|:-----------------------|:-----|
> | sqlmanagedinstance.cpu.utilization.percentage | Average CPU percentage | %    |
> | sqlmanagedinstance.cpu.virtualcores.count     | Virtual core count     |      |

Reviewer: ```suggestion
| Metric Name                                   | Description            | Unit  |
|:----------------------------------------------|:-----------------------|:------|
| sqlmanagedinstance.cpu.utilization.percentage | Average CPU percentage | %     |
| sqlmanagedinstance.cpu.virtualcores.count     | Virtual core count     | Count |
```

<!--Storage-->

> | Metric Name                                     | Description            |
> |:------------------------------------------------|:-----------------------|
> | sqlmanagedinstance.storage.space.reserved.count | Storage space reserved |
> | sqlmanagedinstance.storage.space.used.count     | Storage space used     |

Reviewer: ```suggestion
| Metric Name                                     | Description            | Unit  |
|:------------------------------------------------|:-----------------------|:------|
| sqlmanagedinstance.storage.space.reserved.count | Storage space reserved | Count |
| sqlmanagedinstance.storage.space.used.count     | Storage space used     | Count |
```

<!--Diskio-->

> | Metric Name                            | Description       | Unit  |
> |:---------------------------------------|:------------------|:------|
> | sqlmanagedinstance.bytes.read.bytes    | IO bytes read     | bytes |
> | sqlmanagedinstance.bytes.written.bytes | IO bytes written  | bytes |
> | sqlmanagedinstance.io.requests.count   | IO requests count |       |

Reviewer: ```suggestion
| Metric Name                            | Description       | Unit  |
|:---------------------------------------|:------------------|:------|
| sqlmanagedinstance.bytes.read.bytes    | IO bytes read     | B     |
| sqlmanagedinstance.bytes.written.bytes | IO bytes written  | B     |
| sqlmanagedinstance.io.requests.count   | IO requests count | Count |
```

| X         | AZURERESOURCE     | Id of the SQL Managed Instance instance |

<!--END_DOCUSAURUS_CODE_TABS-->

## FAQ

### How to check in the CLI that the configuration is OK and what are the main options for ?

Once the Plugin installed, log into your Centreon Poller CLI using the *centreon-engine* 
user account and test the Plugin by running the following command:

```bash
/usr/lib/centreon/plugins/centreon_azure_database_sqlmanagedinstance_api.pl \
    --plugin=cloud::azure::database::sqlmanagedinstance::plugin \
    --mode=cpu \
    --custommode=api \
    --subscription='xxxxxxxxx' \
    --tenant='xxxxxxxxx' \
    --client-id='xxxxxxxxx' \
    --client-secret='xxxxxxxxx' \
    --resource='SQL01' \
    --resource-group='xxxxxxxxx' \
    --timeframe='900' \
    --interval='PT5M' \
    --aggregation='Average' \
    --warning-cpu-average='80' \
    --critical-cpu-average='90'
```

Expected command output is shown below:

```bash
OK: Instance 'SQL01' Statistic 'average' Metrics Average CPU percentage: 55.00, Virtual core count: 2 | 'SQL01~average#qlmanagedinstance.cpu.utilization.percentage'=55.00%;0:80;0:90;0; 'SQL01~average#sqlmanagedinstance.cpu.virtualcores.count'=2;;;0; 
```

> The command above checks the *jobs* of an Azure *SQL Managed Instance* instance using the 'api' custom-mode

Reviewer: ```suggestion
The command above checks the *CPU* usage of an Azure *SQL Managed Instance* instance using the 'api' custom-mode
```

(```--plugin=cloud::azure::database::sqlmanagedinstance::plugin --mode=cpu --custommode=api```).
This SQL Managed Instance instance is identified by its id (```--resource='SQL01'```) and the authentication parameters
to be used with the custom mode are specified in the options (```--subscription='xxxxxxxxx' --tenant='xxxxxxx'
--client-id='xxxxxxxx' --client-secret='xxxxxxxxxx'```).

The calculated metrics are the average (```--aggregation='average'```) of values on a 900 secondes / 15 min period (```--timeframe='900'```)
with one sample per 5 minutes (```--interval='PT5M'```).

> This command would trigger a WARNING alarm if the *total jobs* count is reported as over 80
> (```--warning-cpu-average='80'```) and a CRITICAL alarm over 90 (```--critical-cpu-average='90'```).

Reviewer: ```suggestion
This command would trigger a WARNING alarm if the *CPU* usage of the instance is reported as over 80% of its capacity
(```--warning-cpu-average='80'```) and a CRITICAL alarm over 90% (```--critical-cpu-average='90'```).
```
